File: api/router/validation.go
// Copyright 2023 Anass Bouassaba.
//
// Use of this software is governed by the Business Source License
// included in the file licenses/BSL.txt.
//
// As of the Change Date specified in that file, in accordance with
// the Business Source License, use of this software will be governed
// by the GNU Affero General Public License v3.0 only, included in the file
// licenses/AGPL.txt.

package router

import "github.com/kouprlabs/voltaserve/api/service"

func IsValidSortBy(value string) bool {
	return value == "" ||
		value == service.SortByName ||
		value == service.SortByKind ||
		value == service.SortBySize ||
		value == service.SortByEmail ||
		value == service.SortByFullName ||
		value == service.SortByVersion ||
		value == service.SortByFrequency ||
		value == service.SortByDateCreated ||
		value == service.SortByDateModified
}

func IsValidSortOrder(value string) bool {
	return value == "" || value == service.SortOrderAsc || value == service.SortOrderDesc
}

File: ui/src/lib/validation.ts
// Copyright (c) 2023 Anass Bouassaba.
//
// Use of this software is governed by the Business Source License
// included in the file LICENSE in the root of this repository.
//
// As of the Change Date specified in that file, in accordance with
// the Business Source License, use of this software will be governed
// by the GNU Affero General Public License v3.0 only, included in the file
// AGPL-3.0-only in the root of this repository.
import * as Yup from 'yup'

export class YupSchemaFactory {
  static password(field: string) {
    return Yup.string()
      .required(`${field} is required.`)
      .matches(
        /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,})/,
        'Must contain at least 8 characters, one uppercase, one lowercase, one number and one special character.',
      )
  }
}
